File: OneLine/3.7/12.intToRoman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    var val=[1000,900,500,400,100,90,50,40,10,9,5,4,1];
    var roma = ["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"];
    var res = "";
    for (var i=0;i<13;)
    {
        if (num-val[i]>=0)
        {
            num = num-val[i];
            res = res+roma[i];
        }
        else i++;
    }
    return res;
};

File: OneLine/3.7/12.intToRoman.cpp
class Solution {
public:
    int val[13]={1000,900,500,400,100,90,50,40,10,9,5,4,1};
    string roma[13] = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
    string res = "";
    string intToRoman(int num) {
        for (int i=0;i<13;i++)
        {
            while (num>=val[i])
            {
                num = num-val[i];
                res = res+ roma[i];
            }
        }
        return res;
    }
};
